compiler/src/iree/compiler/Codegen/Common/ReconcileTranslationInfo.cpp

```diff
@@ -364,6 +374,191 @@ resolveWorkgroupForAll(RewriterBase &rewriter, FunctionOpInterface funcOp,
   return success();
 }
 
+static LogicalResult
+lowerSplitReductionModifierOp(RewriterBase &rewriter,
+                              FunctionOpInterface entryPointFn,
+                              IREE::Codegen::WorkgroupId delinearizeFrom,
+                              OpFoldResult splitReductionFactor) {
+  std::optional<IREE::HAL::ExecutableExportOp> exportOp =
+      getEntryPoint(entryPointFn);
+  if (!exportOp) {
+    // not entry point.
+    return success();
+  }
+  Block *body = exportOp->getWorkgroupCountBody();
+  if (!body) {
+    return success();
+  }
+  auto splitReduceModifiers = body->getOps<
+      IREE::TensorExt::DispatchWorkgroupCountSplitReductionModifierOp>();
+  if (splitReduceModifiers.empty()) {
+    // Nothing to do.
+    return success();
+  }
+  if (!llvm::hasSingleElement(splitReduceModifiers)) {
+    return exportOp->emitOpError(
+        "unexpected multiple "
+        "iree_tensor_ext.dispatch.workgroup.splitk_modifier");
+  }
+  IREE::TensorExt::DispatchWorkgroupCountSplitReductionModifierOp
+      splitReduceModifier = *splitReduceModifiers.begin();
+  Location loc = splitReduceModifier->getLoc();
+
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(splitReduceModifier);
+  FailureOr<SmallVector<OpFoldResult>> materializedWorkgroupCount =
+      materializeWorkgroupCountComputation(rewriter, entryPointFn,
+                                           splitReductionFactor,
+                                           splitReduceModifier.getWorkload());
+  if (failed(materializedWorkgroupCount)) {
+    return splitReduceModifier->emitOpError(
+        "failed to materialize workgroup count computation in the entry point");
+  }
+
+  SmallVector<OpFoldResult> replacement =
+      llvm::map_to_vector(splitReduceModifier.getSourceWorkgroupCount(),
+                          [](Value v) -> OpFoldResult { return v; });
+  replacement[static_cast<uint64_t>(delinearizeFrom)] =
+      IREE::LinalgExt::mulOfrs(
+          rewriter, splitReduceModifier.getLoc(),
+          replacement[static_cast<uint64_t>(delinearizeFrom)],
```

Comment:
I believe `to_underlying` is the expected way to convert an enum to the underlying integer.
```suggestion
          replacement[llvm::to_underlying(delinearizeFrom)],
```